### PR TITLE
Document tile selection algorithm

### PR DESCRIPTION
**Pull request by kring**
This is a PR into #1032 so merge that first.
This is not a PR into #1022, but it does incorporate those changes. So merge that first, too.
This partially addresses #981.

There's definitely more to say about the tile selection algorithm, but I'm opening this PR now because:

1. I think it's useful even with just what's written so far.
2. It'll be good to get some feedback on this before writing the rest of the sections in the same spirit.

In addition to the new document, this PR lightly refactors a few of the bits of code in the selection algorithm, in order to make them clearer and easier to write about. There shouldn't be any functional changes.

### CODE REVIEW

File: doc/CMakeLists.txt [hunk 1]
```diff
@@ -85,7 +85,10 @@ if(DOXYGEN_FOUND)
     set(DOXYGEN_BUILTIN_STL_SUPPORT YES)
     set(DOXYGEN_USE_MDFILE_AS_MAINPAGE "../README.md")
     set(DOXYGEN_EXCLUDE_PATTERNS "*/node_modules/*")
-    set(DOXYGEN_IMAGE_PATH "./")
+
+    set(DOXYGEN_IMAGE_PATH "")
```
Comment (azrogers) [inline doc/CMakeLists.txt:89]: Is there a reason to set it to an empty string before appending?

**Comment by azrogers** [comment]
Looks good! Just a few questions.

File: doc/topics/selection-algorithm-details.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+The answer is that refinement decisions for these tiles work the same way as for regular tiles, but are based on the value of the [culledScreenSpaceError](@ref Cesium3DTilesSelection::TilesetOptions::culledScreenSpaceError) property instead of `maximumScreenSpaceError`. This allows us to use a lower LOD (larger maximum SSE value) for tiles outside the view frustum or far off in the foggy distance. At an extreme, we can set the [enforceCulledScreenSpaceError](@ref Cesium3DTilesSelection::TilesetOptions::enforceCulledScreenSpaceError) property to `false` if we want to _never_ refine these tiles. This is equivalent to setting `culledScreenSpaceError` to a very large value. The non-visible tiles are selected, so all parts of the model will be represented at some level-of-detail, but the non-visible parts will use the lowest LOD that they can without having a hole in the model.
```
[the remark this answers is not in the document]
Comment (kring) [reply doc/topics/selection-algorithm-details.md:55]: Yeah, that's right. The logic is that it's often (usually?) useful to render non-visible tiles at a lower LOD. If the user spins around quick, you want to have something to show there, but it doesn't have to be the full LOD. Or if the player takes a step backward, you don't want them to fall through the Earth, but you also don't need a perfect-fidelity model if what they're stepping on if it's not visible. Rendering culled tiles at full LOD gets pretty expensive.

File: Cesium3DTilesSelection/src/Tileset.cpp [hunk 3]
```diff
@@ -1270,6 +1271,12 @@ Tileset::_checkOcclusion(const Tile& tile, const FrameState& frameState) {
   return TileOcclusionState::NotOccluded;
 }
 
+namespace {
+
+enum class VisitTileAction { Render, Refine };
```
[the remark this answers is not in the document]
Comment (kring) [reply Cesium3DTilesSelection/src/Tileset.cpp:1276]: Yep, totally just for clarity.

File: doc/CMakeLists.txt (hunk 1 above)
Comment (kring) [reply doc/CMakeLists.txt:89]: Nope! The C++ developer in me feels like that's an uninitialized read, but I checked the cmake docs and `list(APPEND)` treats a non-existent list is empty. I'll change it.

**Comment by kring** [comment]
This is ready for another look.

**Comment by azrogers** [comment]
Thanks for writing this documentation, @kring!

**Pull request merged**